src/appleseed/renderer/kernel/lighting/directlightingintegrator.cpp

```diff
@@ -670,13 +676,32 @@ void DirectLightingIntegrator::add_non_physical_light_sample_contribution(
     // Evaluate the light.
     Vector3d emission_position, emission_direction;
     Spectrum light_value(Spectrum::Illuminance);
-    light->evaluate(
+
+    // Generate a uniform sample in [0,1).
+    sampling_context.split_in_place(2, 1);
+    const Vector2d s = sampling_context.next2<Vector2d>();
+    float probability;
+
+    light->sample(
         m_shading_context,
         sample.m_light_transform,
         m_material_sampler.get_point(),
+        s,
         emission_position,
         emission_direction,
-        light_value);
+        light_value,
+        probability
+    );
+
+    //sample.m_probability *= probability;
```

Comment:
Ah, I think I know the problem. The sun's "intensity" we calculate assume that the sun is a pure directional light, i.e. it's the full sun's energy, not the energy per unit of area. So in a sense, _not_ dividing the total contribution by the probability of the sampled direction is correct. To be correct in the general case however, we must do that division (i.e. uncomment the light above, or better: divide by `probability` when computing the overall weight on line 736) and at the same time, we need to divide the sun's intensity by the disk's surface area. These two computations will cancel out, but we need to do it for other light models where they won't necessarily cancel out.